src/data/list/chain.lean

```diff
@@ -203,4 +204,49 @@ begin
     },
 end
 
+/--
+If `a` and `b` are related by the reflexive transitive closure of `r`,
+then there is a chain starting from `a` and ending on `b`./
+
+NB: This is easier to express with `list.chain'` and `head_of_ne_nil` than
+`list.chain` because of the `list.last` required for the end.
+-/
+lemma exists_zigzag {r : α → α → Prop} {a b : α} (h : relation.refl_trans_gen r a b) :
+  ∃ (l : list α), chain' r l ∧ ∃ (hl : l ≠ nil), head_of_ne_nil l hl = a ∧ last l hl = b :=
```

Comment:
I think we better try to follow mathlib naming rules here. My best guess is that this should be `exists_chain_of_relation_refl_trans_gen`.

Reply:
Much better name, thanks!